### PR TITLE
feat: [CCM-7646]: Implemented sorting, searching and filtering for AS rules and savings data

### PR DESCRIPTION
**Pull request by rishik30**
##### Screenshots:
Screenshot 2022-05-26 at 4 50 19 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by rishik30** [comment]
retrigger titlecheck

File: src/modules/75-ce/components/COGatewayList/SpendVsSavingsChart.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+import css from './COGatewayCumulativeAnalytics.module.scss'
+
+interface SpendVsSavingsChartProps {
+  data?: CumulativeSavings
+  loading: boolean
+  mode?: RulesMode
+  timeRange: TimeRangeFilterType
+}
+
+const toFixedDecimalNumber = (num: number, decimalPlaces = 2) => Number(num.toFixed(decimalPlaces))
```
Comment (rajarshee-harness) [inline src/modules/75-ce/components/COGatewayList/SpendVsSavingsChart.tsx:28]: We already have this in `src/modules/75-ce/utils/convertNumberToFixedDecimalPlaces.ts`

File: src/modules/75-ce/components/COGatewayList/__tests__/__snapshots__/COGatewayList.test.tsx.snap [hunk 2]
```diff
@@ -134,6 +134,147 @@ exports[`Test COGatewayList able to search through the list 1`] = `
           </span>
         </button>
       </div>
+      <div
```
Comment (rajarshee-harness) [inline src/modules/75-ce/components/COGatewayList/__tests__/__snapshots__/COGatewayList.test.tsx.snap:137]: This snapshot file is of 7.7k lines, maybe snapshot tests can replaced with DOM assertions in future.

**Review by rajarshee-harness — DISMISSED** [review]

File: src/modules/75-ce/components/COGatewayList/__tests__/__snapshots__/COGatewayList.test.tsx.snap (hunk 2 above)
Comment (rishik30) [reply src/modules/75-ce/components/COGatewayList/__tests__/__snapshots__/COGatewayList.test.tsx.snap:137]: sure. Thanks for pointing it out.

File: src/modules/75-ce/components/COGatewayList/SpendVsSavingsChart.tsx (hunk 1 above)
Comment (rishik30) [reply src/modules/75-ce/components/COGatewayList/SpendVsSavingsChart.tsx]: done

**Comment by rishik30** [comment]
retrigger jest

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**